File: app/api/v1/admin/logs.py
from datetime import datetime

from fastapi import APIRouter, Query

from app.api.deps import AdminUser, SettingsDep
from app.schemas.admin_logs import LogQueryResult
from app.schemas.common import ApiResponse
from app.services.admin_logs import AdminLogsService

router = APIRouter()


@router.get("/logs", response_model=ApiResponse[LogQueryResult])
async def query_logs(
    _admin: AdminUser,
    settings: SettingsDep,
    page: int = Query(1, ge=1),
    page_size: int = Query(50, ge=1, le=200),
    since: datetime | None = None,
    until: datetime | None = None,
    request_id: str | None = None,
    level: str | None = None,
    q: str | None = None,
) -> ApiResponse[LogQueryResult]:
    data = await AdminLogsService(settings).query_logs(
        since=since,
        until=until,
        request_id=request_id,
        level=level,
        q=q,
        page=page,
        page_size=page_size,
    )
    return ApiResponse(data=data)

File: admin/src/api/logs.ts
import { apiFetch } from './client'
import type { LogQueryResult } from '../types/api'

export interface LogQueryParams {
  page?: number
  page_size?: number
  since?: string
  until?: string
  request_id?: string
  level?: string
  q?: string
}

export function fetchLogs(params: LogQueryParams = {}): Promise<LogQueryResult> {
  const search = new URLSearchParams()
  if (params.page) search.set('page', String(params.page))
  if (params.page_size) search.set('page_size', String(params.page_size))
  if (params.since) search.set('since', params.since)
  if (params.until) search.set('until', params.until)
  if (params.request_id) search.set('request_id', params.request_id)
  if (params.level) search.set('level', params.level)
  if (params.q) search.set('q', params.q)
  const qs = search.toString()
  return apiFetch<LogQueryResult>(`/api/v1/admin/logs${qs ? `?${qs}` : ''}`)
}
